src/talk/renderer/UserStatus/components/UserStatusForm.vue

```diff
@@ -0,0 +1,187 @@
+<!--
+  - @copyright Copyright (c) 2024 Grigorii Shartsev <me@shgk.me>
+  -
+  - @author Grigorii Shartsev <me@shgk.me>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<script setup>
+import { computed, ref } from 'vue'
+import { storeToRefs } from 'pinia'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+import { translate as t } from '@nextcloud/l10n'
+import { useUserStatusStore } from '../userStatus.store.js'
+import UserStatusFormClearAt from './UserStatusFormClearAt.vue'
+import UserStatusFormCustomMessage from './UserStatusFormCustomMessage.vue'
+import UserStatusFormStatusType from './UserStatusFormStatusType.vue'
+import { convertPredefinedStatusToUserStatus } from '../userStatus.utils.js'
+import UserStatusFormPredefinedOption from './UserStatusFormPredefinedOption.vue'
+
+const emit = defineEmits(['submit'])
+
+const userStatusStore = useUserStatusStore()
+
+/** @type {import('vue').Ref<import('./userStatus.types.ts').UserStatus>} */
+const userStatus = ref({ ...userStatusStore.userStatus })
+
+const backupStatus = ref(userStatusStore.backupStatus ? { ...userStatusStore.backupStatus } : null)
+
+const isDirty = ref(false)
+
+const { predefinedStatuses } = storeToRefs(userStatusStore)
```

Comment:
Should be valid as well?

```suggestion
const { userStatus, predefinedStatuses } = storeToRefs(userStatusStore)
```

Reply:
No, we need a copy of the store state here. So it can be changed and then saved or not saved.